doc/Client-Portal Protocol.md

```diff
@@ -542,24 +560,117 @@ This a format for client reconnects.
        sdpMLineIndex: number(mLineIndex),   // optional in RemovedCandidatesMessage
        candidate: string(candidateSdp)
       }
+
+    object(TransportOptions)::
+      {
+        type: "webrtc" | "quic-p2p",
+        id: string(transportId) | null,  // null will result to create a new transport channel.
+      }
+
+    object(P2PQuicParametersMessage)::
+      {
+       type: "quic-p2p-parameters",
+       clientTransportParameters: object(P2PQuicClientParametersMessage) | undefined,
+       serverTransportParameters: object(P2PQuicServerParametersMessage) | undefined
+      }
+
+    object(P2PQuicClientParametersMessage)::
+      {
+        quicKey: arrayBuffer(Key),
+        iceParameters: object(IceParameters)
+      }
+    object(P2PQuicServerParametersMessage)::
+      {
+        iceParameters: object(IceParameters)
+      }
+
+    object(IceParameters)::
+      {
+        usernameFragment: string(UsernameFragment),
+        password: string(Password)
+      }
 **ResponseData**: undefined if **ResponseStatus** is “ok”.
 ### 3.3.15 Participant Receives Session Progress
 **NotificationName**: “progress”<br>
 
-**NotificationData**: The SessionProgress object with following definition.
+**NotificationData**: The SessionProgress or TransportProgress object with following definition.
 
-  object(SessionProgress)::
+  object(TransportProgress)::
```

Comment:
What's the "transportId" for a "webrtc" connection since "TransportId" setting is meaning less for webrtc, "SessionId"?
Then how we distinguish "TransportProgress" and "SessionProgress", cause they are all sent through the notification name "progress"? Are they one thing?
And I have another question, for "quic" connection, "TransportId" and "SessionId" are different? Then for "quic", what's the difference between a "Transport ready status" and a "Session ready status"?

Reply:
Adding a new property `type` to `TransportProgress` and `SessionProgress`? Transport ready means it's able to send or receive data. Session ready means data source or sink is ready? That's my thought about conference server. If session ready is not needed, it's okay for QUIC connections.